core/src/token_info/cached.rs

```diff
@@ -60,14 +74,31 @@ impl TokenInfoCache {
 impl TokenInfoFetching for TokenInfoCache {
     fn get_token_info<'a>(&'a self, id: TokenId) -> BoxFuture<'a, Result<TokenBaseInfo>> {
         async move {
-            if let Some(info) = self.cache.read().await.get(&id).cloned() {
-                return Ok(info);
+            match self.cache.read().await.get(&id) {
+                Some(CacheEntry::TokenBaseInfo(info)) => return Ok(info.clone()),
+                Some(CacheEntry::UnretryableError(reason)) => {
+                    return Err(anyhow!("cached error: {}", reason));
```

Comment:
Maybe it makes sense to use `Context` here?
```suggestion
                    return Err(
                        anyhow!(reason)
                            .context("cached error retrieving token info"),
                    );
```

I'm also fine with not using it, just thought it was a good application for it.